### PR TITLE
To Do list: list structure

### PR DESCRIPTION
**Pull request by Oscah01**
- Set up a new project with webpack 
  - Create an index.js file and set an array of some simple to do tasks (array of objects). 
 - Wrote a function to iterate over the tasks array and populate an HTML list item element for each task.
 - created list of tasks in the dedicated placeholder.
  - Created a style.css and set rules for the To Do List. 
  - added config files

### CODE REVIEW

**Review by OLIPLICHE — COMMENTED** [review]
Hi @Oscah01, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

**You have submitted a project, that does not have Linters installed, and the provided readme template. I would advise you to follow the general requirements of your project**

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by addod19** [comment]
### Hi @Oscah01, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because the README file is missing some important headers. To find the headers, let us kindly check this [link](https://github.com/microverseinc/readme-template) and also go back to the project requirements and let us find the headers that should be present and the optional ones as well

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1 +1,188 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- FAQ (optional) -->
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (1GORDON) [inline README.md]: Great job @Oscah01 🤝

- [ ] [Optional] Kindly fix the broken MIT license link to make your readme file more professional.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Oscah01 ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [ ] Kindly do not commit `node_modules` to GitHub. You may add the node_modules folder to your `.gitignore` file.
- [ ] As stated in the project requirements, could you create an array of some simple to-do tasks (array of objects) with each object containing three keys - description [string], completed [bool], index: [number].

_Check the comments under the review._

## Optional suggestions

- [ ] You may always write your commit messages in an imperative tense something like `Fix Linter Errors` instead of `Fixed Linter Errors`.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: node_modules/@discoveryjs/json-ext/package.json [hunk 2]
```diff
@@ -1,56 +0,0 @@
...
-        "lint": "eslint src test",
-        "lint-and-test": "npm run lint && npm test",
-        "build": "rollup --config",
-        "test:all": "npm run test:src && npm run test:dist",
-        "test:src": "npm test",
-        "test:dist": "cross-env MODE=dist npm test && cross-env MODE=dist-min npm test",
-        "build-and-test": "npm run build && npm run test:dist",
-        "coverage": "c8 --reporter=lcovonly npm test",
-        "prepublishOnly": "npm run lint && npm test && npm run build-and-test"
-    },
```
Comment (Nemwel-Boniface) [inline node_modules/@discoveryjs/json-ext/package.json:35]: - [ ] [OPTIONAL] When starting the project, I was getting an error that says "Missing start scripts". THis was easily fixed by adding the start script to the scripts in your package.json file.

**This is the error:**

image

**This is the updated package.json:**

```
"scripts": {
    "start": "webpack serve --open",
    "dev": "webpack serve",
    "linters": "hint . && stylelint \"**/*.css\" && eslint .",
    "build": "webpack --mode=production --node-env=production",
    "build:dev": "webpack --mode=development",
    "build:prod": "webpack --mode=production --node-env=production",
    "watch": "webpack --watch"
  },
```

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @Oscah01          :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: node_modules/@discoveryjs/json-ext/package.json (hunk 2 above)
Comment (Oscah01) [reply node_modules/@discoveryjs/json-ext/package.json:35]: Thank you alot

**Pull request merged**